src/utils/polygon.cpp

```diff
@@ -156,6 +156,89 @@ Polygons PolygonRef::offset(int distance, ClipperLib::JoinType joinType, double
     return ret;
 }
 
+Polygon Polygons::convexHull() const
+{
+    // Implements Andrew's monotone chain convex hull algorithm
+    // See https://en.wikibooks.org/wiki/Algorithm_Implementation/Geometry/Convex_hull/Monotone_chain
+
+    using IntPoint = ClipperLib::IntPoint;
+
+    std::vector<IntPoint> all_points;
+    for (const ClipperLib::Path &path : paths)
+    {
+        // Reserve big enough for result, but also maintain O(1)
+        // amortized resize cost
+        size_t reserve_size = std::max(all_points.size() + path.size(),
+                                       2*all_points.size());
```

Comment:
I think std::vector already does that. http://www.cplusplus.com/reference/vector/vector/reserve/
"the function causes the container to reallocate its storage increasing its capacity to n (or greater)."

Why don't you just reserve once and compute the amount of points needed in a separate loop?
```
reserve_size = 0;
for (Path path : paths)
    reserve_size += path.size();
all_points.reserve(reserve_size);
```